tests/tests.rs

```diff
@@ -831,6 +831,17 @@ sherlock\x00can extract a clew from a wisp of straw or a flake of cigar ash;
     assert_eq!(lines, expected);
 });
 
+// See: https://github.com/BurntSushi/ripgrep/issues/109
+clean!(max_depth, "far", ".", |wd: WorkDir, mut cmd: Command| {
+    wd.create_dir("one");
+    wd.create_dir("one/too");
+    wd.create("one/too/many", "far");
+
+    cmd.arg("--maxdepth").arg("1");
+
+    wd.assert_err(&mut cmd);
```

Comment:
The count applies to whether to descend into a directory or not. At depth `0`, you don't descend ever, you just yield whatever was given. So a walkdir of `./` at depth `0` yields `./`. At depth `1`, you descend once, so that a walkdir of `./` at depth `1` yields `one`, but doesn't descend into `one`. Finally, at depth `2`, a walkdir of `./` yields `./one` and then `./one/too` and also `./one/foo`, but doesn't descend into `./one/too`.